src/software/thunderscope/binary_context_managers/game_controller.py

```diff
@@ -21,51 +21,80 @@
 
 
 class Gamecontroller(object):
-    """ Gamecontroller Context Manager """
+    """Gamecontroller Context Manager"""
 
     CI_MODE_LAUNCH_DELAY_S = 0.3
     REFEREE_IP = "224.5.23.1"
     CI_MODE_OUTPUT_RECEIVE_BUFFER_SIZE = 9000
 
-    def __init__(self, supress_logs: bool = False) -> None:
-        """Run Gamecontroller
+    def __init__(
+        self,
+        supress_logs: bool = False,
+        use_unconventional_port: bool = False,
+        not_launch_gc: bool = True,
+    ) -> None:
+        """
+        Run Gamecontroller
 
         :param supress_logs: Whether to suppress the logs
+        :disregard_gamecontroller_port: if set to True, the gamecontroller port would be set to 12393, else it would highly likely be 40000, i.e. the default port that everybodies uses!
+        :not_launch_gc Whether to launch the gamecontroller or not!
         """
+
         self.supress_logs = supress_logs
+        self.not_launch_gc = not_launch_gc
+
+        if use_unconventional_port:
+            # the intention of setting this port is to ensure that we don't listen other people's
+            # gamecontroller. It is highly unlikely that other people would use port 12393 since
+            # this is a random number that was set by us!
+            self.referee_port = self.next_free_port(start_port=12393)
```

Comment:
can you make this a random int above 1024 and below 65535? If multiple people are running thunderscope_main, we can conflict with ourselves